Cython/Compiler/Symtab.py

```diff
@@ -931,6 +931,14 @@ def declare_cfunction(self, name, type, pos,
             var_entry.scope = entry.scope
             entry.as_variable = var_entry
         type.entry = entry
+        pos_filetype = pos[0]._file_type if pos else None
+        if (type.exception_check and type.exception_value is None and type.nogil and
+                not pos[0].in_utility_code and
+                defining and  # don't warn about external functions here - the user likely can't do anything
+                pos_filetype != 'pxd'  # again, exclude extern functions (is_pxd is unreliable here)
```

Comment:
1) Does this also exclude inline functions in .pxd files?
2) Why is `is_pxd` unreliable?

Reply:
> 1. Does this also exclude inline functions in .pxd files?

Yes - intentionally (I'm working on the basis that anything in a pxd is likely something the user doesn't control, and that they get a message when they use it anyway)

> 2. Why is is_pxd unreliable?

Not sure off the top of my head, but it didn't seem to be set in the example that I tested. I'll go back and see if I can fix that though.

(Will also come back to the other two comments later)

Reply:
It looks like `in_pxd` means that Cython generates code to cimport the function by declaring function pointers (which conflicts with `inline`). I've had a go at trying to fix that so it's probably a bit better.